File: tests/fixtures/openai-realtime/openai-realtime-send-safety-identifier-fixed/01-header-present.ts
import WebSocket from 'ws';

function setup(apiKey: string, callerLanguage: string): WebSocket {
  const url = 'wss://api.openai.com/v1/realtime?model=gpt-realtime';
  return new WebSocket(url, {
    headers: {
      Authorization: `Bearer ${apiKey}`,
      'OpenAI-Safety-Identifier': hashCallerLanguage(callerLanguage),
    },
  });
}

declare function hashCallerLanguage(value: string): string;

File: tests/fixtures/openai-realtime/openai-realtime-send-safety-identifier-fixed/01_header_present.py
import os

import websockets

WS_URL = "wss://api.openai.com/v1/realtime?model=gpt-realtime"


async def connect(hashed_account_id):
    return await websockets.connect(
        WS_URL,
        additional_headers={
            "Authorization": f"Bearer {os.environ.get('OPENAI_API_KEY')}",
            "OpenAI-Safety-Identifier": hashed_account_id,
        },
    )
